### PR TITLE
Add UI pop-up fixes

### PR DESCRIPTION
**Pull request by HakimSkaik**
Creating a new pr for #307 since it had docker files pushed as well

image
image

### CODE REVIEW

File: myApp/app/components/BuildingPopUp.tsx [hunk 1]
```diff
@@ -87,4 +128,93 @@ export const BuildingPopup: React.FC<BuildingPopupProps> = ({
   );
 };
 
-export default BuildingPopup;
+// Inline Styles
+const styles = StyleSheet.create({
```
Comment (sbiothmane) [inline myApp/app/components/BuildingPopUp.tsx:132]: Consider moving the styles outside of the component file.

File: myApp/app/components/ShuttleInfoPopup.jsx [hunk 2]
```diff
@@ -1,74 +1,122 @@
...
+          ) : (
+            <Text style={styles.alertText}>No bus available.</Text>
+          )}
         </View>
       </View>
     </Modal>
   );
 }
 
-// Add PropTypes validation to address SonarQube warnings
```
Comment (sbiothmane) [inline myApp/app/components/ShuttleInfoPopup.jsx:64]: We need Props validation for avoiding bugs and for the code quality pass sonarqube.

File: myApp/app/components/ShuttleInfoPopup.jsx [hunk 3]
```diff
@@ -1,74 +1,122 @@
...
-  onClose: PropTypes.func.isRequired,
-  shuttleDetails: PropTypes.shape({
-    error: PropTypes.string,
-    waitTime: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
-    shuttleRideTime: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
-    totalTime: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
-  }),
-};
+// Inline styles
+const styles = StyleSheet.create({
```
Comment (sbiothmane) [inline myApp/app/components/ShuttleInfoPopup.jsx:66]: Same thing for this styles. Plus it looks like the styles are similar which will trigger duplication so consider using one style file for both pop ups.

File: myApp/app/utils/shuttleUtils.js
Comment (sbiothmane) [inline myApp/app/utils/shuttleUtils.js:1]: Consider adding console.err. where you return null when validating that you got the data so we can know what failed instead of just getting back null.

**Review by sbiothmane — CHANGES_REQUESTED** [review]
Overall great work, just need some code quality improvement:
- seperate the styles to reuse and avoid duplications
- adding so console.err would help maintainability

File: myApp/app/components/BuildingPopUp.tsx (hunk 1 above)
Comment (HakimSkaik) [reply myApp/app/components/BuildingPopUp.tsx]: Thanks! I've refactored the styles out of the component and centralized them in `PopUpStyles.js`, which is now shared between `BuildingPopup` and `ShuttleInfoPopup` to maintain consistency and reduce duplication.

File: myApp/app/components/ShuttleInfoPopup.jsx (hunk 2 above)
Comment (HakimSkaik) [reply myApp/app/components/ShuttleInfoPopup.jsx:64]: Good catch! I’ve added PropTypes validation to `ShuttleInfoPopup.jsx` to ensure prop types are checked properly and to address the SonarQube warnings.

File: myApp/app/components/ShuttleInfoPopup.jsx (hunk 3 above)
Comment (HakimSkaik) [reply myApp/app/components/ShuttleInfoPopup.jsx]: Agreed! I’ve removed the duplicated inline styles from `ShuttleInfoPopup.jsx` and migrated both popup components to use a shared `PopUpStyles.js` file for consistent styling and less code duplication.

File: myApp/app/utils/shuttleUtils.js
Comment (HakimSkaik) [reply myApp/app/utils/shuttleUtils.js:1]: Done! I’ve added console.error logs to every conditional null return in shuttleUtils.js to improve traceability and make debugging easier when schedule data isn’t available.

**Comment by HakimSkaik** [comment]
This pr also address this error:
image

Where I
- Enhanced shuttleUtils.js with descriptive console.error logs for all null/error cases
- Enabled TEST_MODE for fetchShuttleScheduleByDay to support testing at any time
- Updated shuttleUtils.test.js to reflect new error handling

So now the Shuttle To Layola Button is back to expected behaviour:
image

**Review by sbiothmane — APPROVED** [review]
Great fixes!

File: myApp/app/utils/shuttleUtils.js [hunk 4]
```diff
@@ -40,103 +32,67 @@ export const estimateShuttleTravelTime = async (
...
   const stopKey = destinationCampus === "LOY" ? "SGW" : "LOY";
-  const stopSchedule =
-    schedule[stopKey]?.map((time) => {
-      const [hours, minutes] = time.split(":").map(Number);
-      return hours * 60 + minutes; // Convert time to minutes
-    }) || [];
+  const stopSchedule = schedule[stopKey]?.map((time) => {
+    const [hours, minutes] = time.split(":").map(Number);
+    return hours * 60 + minutes;
+  }) || [];
```
[the remark this answers is not in the document]
Comment (HakimSkaik) [reply myApp/app/utils/shuttleUtils.js:61]: > As raised by CodeCov, we should add a test for the case where we initialize the constant with empty [ ]. image

Thanks for pointing this out! I’ve added a test to cover the case where `stopSchedule` initializes as an empty array. The line is now covered in Codecov. Appreciate the catch!

[the remark this answers is not in the document]
Comment (HakimSkaik) [reply myApp/app/utils/shuttleUtils.js:61]: > I also looked at the CodeCov report and noticed some uncovered lines that we can add tests for like this one: image You did a really good job with the tests and you have a good coverage, but I would still suggest for you to look at the coverage report here and see if we can solve some of the uncovered lines https://app.codecov.io/gh/Campus-Compass-SOEN-390/Watermelons_SOEN390/pull/308?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Campus-Compass-SOEN-390

 Great suggestion! I added targeted tests to cover `NaN scenarios `and invalid `shuttleRideTime` calculations. These lines are now also reflected in the coverage report

[the remark this answers is not in the document]
Comment (7anine) [reply myApp/app/utils/shuttleUtils.js:61]: Great job! I just ran the CodeCov Action again on your PR and the coverage is up to 93%.  Looks like we're ready to merge.

**Review by 7anine — APPROVED** [review]
